put operation raises exception: AttributeError: 'NoneType' object has no attribute 'absolute_update'

My test running on `gcsfs-2021.7.0` worked without any issues, though upgrading to `gcsfs-2021.9.0` gave me the following error on the `gcsfs.core.GCSFileSystem.put` method. See the following traceback:

# Traceback
```
damsat/ingestion/ingestion.py:671: in do_cut_general
    move(filename, remote_filename)
../../miniconda3/envs/damsat/lib/python3.7/site-packages/libsat/filesystem.py:84: in move
    fs_target.put(source_path, target_path, recursive=recursive)
../../miniconda3/envs/damsat/lib/python3.7/site-packages/fsspec/asyn.py:91: in wrapper
    return sync(self.loop, func, *args, **kwargs)
../../miniconda3/envs/damsat/lib/python3.7/site-packages/fsspec/asyn.py:71: in sync
    raise return_result
../../miniconda3/envs/damsat/lib/python3.7/site-packages/fsspec/asyn.py:25: in _runner
    result[0] = await coro
../../miniconda3/envs/damsat/lib/python3.7/site-packages/fsspec/asyn.py:422: in _put
    coros, batch_size=batch_size, callback=callback
../../miniconda3/envs/damsat/lib/python3.7/site-packages/fsspec/asyn.py:213: in _run_coros_in_chunks
    results.append(await coro)
../../miniconda3/envs/damsat/lib/python3.7/asyncio/tasks.py:560: in _wait_for_one
    return f.result()  # May raise f.exception().
```

failing on the following function call
```python
    async def _put_file(
        self,
        lpath,
        rpath,
        metadata=None,
        consistency=None,
        content_type="application/octet-stream",
        chunksize=50 * 2 ** 20,
        callback=None,
        **kwargs,
    ):
        # enforce blocksize should be a multiple of 2**18
        if os.path.isdir(lpath):
            return
        consistency = consistency or self.consistency
        checker = get_consistency_checker(consistency)
        bucket, key = self.split_path(rpath)
        with open(lpath, "rb") as f0:
            size = f0.seek(0, 2)
            f0.seek(0)
            if callback is not None:
                callback.set_size(size)
    
            if size < 5 * 2 ** 20:
                await simple_upload(
                    self,
                    bucket,
                    key,
                    f0.read(),
                    consistency=consistency,
                    metadatain=metadata,
                    content_type=content_type,
                )
>               callback.absolute_update(size)
E               AttributeError: 'NoneType' object has no attribute 'absolute_update'
```

# possible related change
Perhaps since the callback was introduced https://github.com/dask/gcsfs/commit/552ccb3735b9bcfe153977ca78fa842b05ad59bf it may be None, although that case is not handled in the code.

--- a/gcsfs/core.py
+++ b/gcsfs/core.py
@@ -904,8 +904,7 @@ class GCSFileSystem(AsyncFileSystem):
         with open(lpath, "rb") as f0:
             size = f0.seek(0, 2)
             f0.seek(0)
-            if callback is not None:
-                callback.set_size(size)
+            callback.set_size(size)
 
             if size < 5 * 2 ** 20:
                 await simple_upload(
@@ -932,8 +931,7 @@ class GCSFileSystem(AsyncFileSystem):
                         self, location, bit, offset, size, content_type
                     )
                     offset += len(bit)
-                    if callback is not None:
-                        callback.absolute_update(offset)
+                    callback.absolute_update(offset)
                     checker.update(bit)
 
             checker.validate_json_response(out)
@@ -1009,12 +1007,11 @@ class GCSFileSystem(AsyncFileSystem):
             timeout=self.requests_timeout,
         ) as r:
             r.raise_for_status()
-            if callback is not None:
-                try:
-                    size = int(r.headers["content-length"])
-                except (KeyError, ValueError):
-                    size = None
-                callback.set_size(size)
+            try:
+                size = int(r.headers["content-length"])
+            except (KeyError, ValueError):
+                size = None
+            callback.set_size(size)
 
             checker = get_consistency_checker(consistency)
             os.makedirs(os.path.dirname(lpath), exist_ok=True)
@@ -1025,8 +1022,7 @@ class GCSFileSystem(AsyncFileSystem):
                         break
                     f2.write(data)
                     checker.update(data)
-                    if callback is not None:
-                        callback.relative_update(len(data))
+                    callback.relative_update(len(data))
 
             validate_response(r.status, data, rpath)  # validate http request
             checker.validate_http_response(r)  # validate file consistency

--- a/gcsfs/core.py
+++ b/gcsfs/core.py
@@ -16,6 +16,7 @@ import weakref
 
 from fsspec.asyn import sync_wrapper, sync, AsyncFileSystem
 from fsspec.utils import stringify_path, setup_logging
+from fsspec.callbacks import NoOpCallback
 from fsspec.implementations.http import get_client
 from .retry import retry_request, validate_response
 from .checkers import get_consistency_checker
@@ -896,6 +897,7 @@ class GCSFileSystem(AsyncFileSystem):
         # enforce blocksize should be a multiple of 2**18
         if os.path.isdir(lpath):
             return
+        callback = callback or NoOpCallback()
         consistency = consistency or self.consistency
         checker = get_consistency_checker(consistency)
         bucket, key = self.split_path(rpath)
@@ -1032,6 +1034,7 @@ class GCSFileSystem(AsyncFileSystem):
 
     async def _get_file(self, rpath, lpath, callback=None, **kwargs):
         u2 = self.url(rpath)
+        callback = callback or NoOpCallback()
         await self._get_file_request(u2, lpath, callback=callback, **kwargs)
 
     def _open(

